### PR TITLE
Structured log: use util/log(logrus) to replace ngaut/log

### PR DESCRIPTION
**Pull request by andelf**
TODO:
- [x] basic function
- [x] vendor update
- [x] ``--log-format`` options
- [ ] util/log testcases

rewrite of #3256

### CODE REVIEW

**Comment by andelf** [comment]
``show status like 'character_set_results';`` fails?

File: util/log/formatter.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	for k, v := range entry.Data {
+		if k != "file" && k != "line" {
+			fmt.Fprintf(b, " %v=%v", k, v)
+		}
+	}
+	b.WriteByte('\n')
+	return b.Bytes(), nil
+}
+
+func stringToLogFormatter(format string, disableTimestamp bool) logrus.Formatter {
```
Comment (hanfei1991) [inline util/log/formatter.go]: Why not move this function to compat.go ?

File: util/log/formatter.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	"bytes"
+	"fmt"
+	"path"
+	"runtime"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// isSKippedPackageName tests wether path name is on log library calling stack.
```
Comment (hanfei1991) [inline util/log/formatter.go]: wether -> whether

File: util/log/formatter.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+// modifyHook injects file name and line pos into log entry.
+type contextHook struct{}
+
+// Fire implements logrus.Hook interface
+// https://github.com/sirupsen/logrus/issues/63
+func (hook *contextHook) Fire(entry *logrus.Entry) error {
+	pc := make([]uintptr, 3)
+	cnt := runtime.Callers(6, pc)
```
Comment (hanfei1991) [inline util/log/formatter.go]: add comments to explain these two lines. It's hard to understand.

**Comment by hanfei1991** [comment]
Can we move ngaut/log out of vendor ?

**Comment by hanfei1991** [comment]
@andelf `show status like 'character_set_results';` sometimes fails. We are fixing it, never mind.

**Comment by tiancaiamao** [comment]
I suggest you split it into three PRs:
1. update vendor to import logrus
2. introduce util/log package that wrap logrus (better with test)
3. use the new log package to replace ngaut/log

**Comment by andelf** [comment]
Will be closed and splitted to 3 PRs.

**Comment by zhexuany** [comment]
@andelf I gonna close this PR since you said you gonna split this into there PRs, but feel free open it later.

**Pull request closed without merging**